sherpa/astro/io/crates_backend.py

```diff
@@ -952,8 +952,7 @@ def get_pha_data(arg, make_copy=True, use_background=False):
         # Chandra Level 3 PHA files
         for ii in range(phadataset.get_ncrates()):
             block = phadataset.get_crate(ii + 1)
-            hduclas2 = block.get_key('HDUCLAS2')
-            if hduclas2 is not None and hduclas2.value == 'BKG':
+            if (_try_hdr_key(block, 'HDUCLAS2') == 'BKG'):
```

Comment:
@jbudynk it would be good to have a test for this change. The issue only shows in CIAO but in a test for an external tool.

There are several options, and some require some infrastructure work, so part of this could be done in a separate PR.

One option is to create an integration test that requires a FITS backend. This is the easiest one, although in this case it could only be tested in CIAO. A better option would be to create a unit test for the crates backend, mock pycrates and make it return byte strings as it does now. This requires some preliminary work similar to what I did for the chips_backend. But even in this case we would need an integration test, too.

In any case we should not fix anything if we don't have a test in place to test the fix. When I review these fixes I checkout the PR, I temporarily remove the fix while keeping the tests, then I make sure that the tests fail. Then I checkout the PR again and run the tests, seeing that they pass.

I know you ran the tests in CIAO directly using an external test, but since this issue was not picked up by a Sherpa test in the first place, we need to add one in Sherpa, as we would with any other bug.